identity/src/main/java/com/android/identity/GattServer.java

```diff
@@ -209,25 +268,103 @@ public void onCharacteristicReadRequest(BluetoothDevice device, int requestId, i
             } catch (SecurityException e) {
                 reportError(e);
             }
+        } else if (mCharacteristicL2CAP != null
+                && characteristic.getUuid().equals(mCharacteristicL2CAPUuid)) {
+            if (!mSupportL2CAP) {
+                reportError(new Error("Unexpected read request for L2CAP characteristic, not supported"));
+                return;
+            }
+            if (mServerSocket == null) {
+                reportError(new Error("Socket for L2CAP not available"));
+                return;
+            }
+            int psm = mServerSocket.getPsm();
+            byte[] psmValue = ByteBuffer.allocate(4).putInt(psm).array();
+            if (mLog.isTransportEnabled()) {
+                mLog.transport("PSM Value generated: " + psm + " byte: " + Util.toHex(psmValue));
+            }
+            mGattServer.sendResponse(device,
+                    requestId,
+                    BluetoothGatt.GATT_SUCCESS,
+                    0,
+                    psmValue);
+            acceptConnection();
```

Comment:
I don't think this is right.... basically you are assuming that the GATT client will try to connect to the L2CAP socket right after reading the characteristic.